Sources/MongoSwift/MongoCursor.swift

```diff
@@ -41,6 +44,17 @@ public class MongoCursor<T: Codable>: Cursor {
         case cached(T?)
         /// Indicates that there is no value cached.
         case none
+
+        /// Get the contents of the cache and clear it.
+        fileprivate mutating func clear() -> T?? {
+            switch self {
+            case let .cached(result):
+                self = .none
+                return result
+            case .none:
+                return nil
+            }
+        }
     }
 
     /// Tracks the caching status of this cursor.
```

Comment:
I decided to leave the cache as part of `MongoCursor` since it applied only to cursors and not change streams, but we could alternatively put the caching logic into `Cursor<CursorKind>` if we wanted to. It might simplify the code in `MongoCursor` a little bit.